File: src/main/java/com/github/sorusclient/client/bootstrap/Launcher.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.bootstrap;

import com.github.sorusclient.client.Sorus;

import java.util.List;

public class Launcher {

    public static void main(String[] args) throws Exception {
        String version = args[List.of(args).indexOf("--minecraftVersion") + 1];
        BootstrapManager.loadJson(Launcher.class.getClassLoader().getResource("sorus.json"), version);

        Sorus.INSTANCE.initialize();

        var main = Class.forName("net.minecraft.client.main.Main");
        var mainMethod = main.getMethod("main", String[].class);
        mainMethod.invoke(null, (Object) args);
    }

}

File: src/main/kotlin/com/github/sorusclient/client/bootstrap/Launcher.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.bootstrap

import com.github.sorusclient.client.Sorus

class Launcher {

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            val minecraftVersionIndex = args.indexOf("--minecraftVersion") + 1
            val version = args[minecraftVersionIndex]
            BootstrapManager.loadJson(Launcher::class.java.classLoader.getResource("sorus.json")!!, version)

            Sorus.initialize()

            val main = Class.forName("net.minecraft.client.main.Main")
            val mainMethod = main.getMethod("main", Array<String>::class.java)

            val args2 = Array<String?>(args.size - 2) { null }
            System.arraycopy(args, 0, args2, 0, minecraftVersionIndex - 1)
            System.arraycopy(args, minecraftVersionIndex + 1, args2, minecraftVersionIndex - 1, args.size - minecraftVersionIndex - 1)
            mainMethod.invoke(null, args2)
        }

    }

}